src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -209,6 +232,24 @@ impl Vcpu {
         self.kvm_vcpu.peripherals.mmio_bus = Some(mmio_bus);
     }
 
+    /// Attaches the fields required for debugging
+    #[cfg(feature = "debug")]
+    pub fn attach_debug_info(&mut self, gdb_event: Sender<usize>) {
+        self.gdb_event = Some(gdb_event);
+    }
+
+    /// Uses the kvm api to translate the virtual address
+    #[cfg(feature = "debug")]
+    fn translate_gva(&self, address: u64) -> Option<NonZeroU64> {
+        let tr = self.kvm_vcpu.fd.translate_gva(address).unwrap();
+
+        if tr.valid == 0 {
+            return None;
+        }
+
+        NonZeroU64::new(tr.physical_address)
+    }
```

Comment:
Mh, you're right that translate_gva cannot failt (the only error it might return is EFAULT if you pass in a nonsense `kvm_translate` pointer, but kvm-ioctls ensures we don't do that), but we can't really rely on that I think.

The other thing is that I don't immediately see they we assume that the result of the gva->gpa translation will never return 0 (we convert a 0 gpa into `None` via `NonZeroU64::new`). But I also don't understand why we're using `NonZeroU64` in the first place. At the call site, we just convert it back into a u64, so I think we can just keep it a u64 all the way through. However, _after_ converting to back to `u64` in `FirecrackerTarget::translate_gva`, we have a special case in `FirecrackerTarget::write_addrs` that checks whether the gpa is 0 (which it cannot be because of the `NonZeroU64` dance).

So how about we change this function be something like 

```suggestion
    fn translate_gva(&self, address: u64) -> Result<u64, KvmTranslateError> {
        let tr = self.kvm_vcpu.fd.translate_gva(address)?;

        if tr.valid == 0 {
            return Err(KvmTranslateError::NoTranslation);
        }

        Ok(tr.physical_address)
    }
```